src/Services/PatientService.php

```diff
@@ -113,45 +143,19 @@ public function getFreshPid()
     public function insert($data)
     {
         $fresh_pid = $this->getFreshPid();
+        $data['pid'] = $fresh_pid;
+        $data['pubpid'] = $fresh_pid;
+        $data['date'] = date("Y-m-d H:i:s");
+        $data['regdate'] = date("Y-m-d H:i:s");
 
-        $sql = " INSERT INTO patient_data SET";
-        $sql .= "     pid=?,";
-        $sql .= "     title=?,";
-        $sql .= "     fname=?,";
-        $sql .= "     mname=?,";
-        $sql .= "     lname=?,";
-        $sql .= "     street=?,";
-        $sql .= "     postal_code=?,";
-        $sql .= "     city=?,";
-        $sql .= "     state=?,";
-        $sql .= "     country_code=?,";
-        $sql .= "     phone_contact=?,";
-        $sql .= "     DOB=?,";
-        $sql .= "     sex=?,";
-        $sql .= "     race=?,";
-        $sql .= "     ethnicity=?";
+        $query = $this->buildInsertColumns($data);
```

Comment:
Hello @sjpadgett  Sir, I just noticed something buggy here if we pass any column from `patient_data` table that field will get inserted regardless of what want I mean there should be a limit of the field which are allowed to the user. User should not be able to modify every field in the Database.

Reply:
I will open a particular issue for this if you confirm 👍

Reply:
First, that's the controller's job! The services should exposed everything possible. 
The service should validate data integrity for sure but, what gets to the service will depend on the API audience and their role and permissions. With some folks we may want to allow only reads while others, full access.
This is especially true for our core services where we need services data exposure to be fully exposed for tasks such as form management.
What happens when two years down the road someone writes a controller that wants to read/update a field you deem inappropriate now. Or a new field gets added to a table, do you want me calling you up to come patch hard coded fields in code?

In the case of building queries for database transactions, that function should be kept pure and simple with its task. 

You're right to question this though so, you're thinking like an engineer. But, buildInsertColumns() is designed to dynamically build, not qualify. That duty should happen before getting there. We probably need a permissions function where we can whitelist columns based on API audiences logged in role/scope. Now that is worthy of an issue...

Reply:
Thanks for clearing that up the only thing bothering me was accessibility. Whitelisting based on permission sounds good 👍

Reply:
yep, that's why I make the big bucks of $0.25 an hour.